the-collab-lab/tcl-20-whats-near-me
# Create a table of nearby destinations on the List page

## Description

- User can switch to the List tab to see a table of all of the locations currently populated on the map
- We created a table from scratch using built-in table functionality. We tried a few table plugins first, but they seemed like an overly complex solution to a simple program, so we scaled back.
- The list should include the name of the location, a description, a link to the Wikipedia page, and a thumbnail if it's available (we adjusted the thumbnail size to 50px x 50px because it seemed to fit the table better)
- The list should be scrollable, while the tabs link at the bottom stays put
- We also changed the location to New Orleans city center so there can be some more data on the map / list, and New Orleans is just more fun :)

## Related Issue

Story: Integrate with "near me" API

## Acceptance Criteria

When the app starts, a call should be made to the proxy endpoint using the latitude and longitude at the center of the map

The returned data should be used to render a table of information (sorted by distance) on the /list route that includes the following:

Distance (in km)
Thumbnail (if present, scaled down to 24px x 24px using object-fit: cover) 
Name
Description
A link to view the place on Wikipedia

Links out to Wikipedia should use the pageid property that is returned by the API, using the following format: https://en.wikipedia.org/?curid=<pageid> (e.g., https://en.wikipedia.org/?curid=60401407)


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates
- We gave the bottom nav bar a background color (white) so the text doesn't overlap and deleted the funky padding.

### Before



### After

![screen grab of list page in Explore Your World app](https://user-images.githubusercontent.com/44718574/105789320-11b75680-5f37-11eb-8d8f-83f8a86f2379.png)

## Testing Steps / QA Criteria

1. Open the app and see the populated map page based in New Orleans, LA.
2. Navigate to the list page
3. See a list of all of the locations that show up on the map!
4. Name and link should always be populated, description and thumbnail will show up if available, if not those sections will be blank.
5. Decide where you want to go when you visit New Orleans!
